### ISSUE TITLE
File Type References

### ISSUE THREAD

**Issue by devinross** [sanity-io/swift-sanity#23]
While the support for Image types is there, the is no support for "File" types in the SDK. I've implemented this into my own project to handle files, but would be nice to have this directly in the sdk.

```
import Sanity


public extension SanityType {
	struct File {
		public let asset: Ref
		public let _type: String

		public init(asset: Ref, _type: String) {
			self.asset = asset
			self._type = _type
		}
	}
}


extension SanityType.File: Decodable {
	enum CodingKeys: String, CodingKey {
		case _type, asset
	}

	public init(from decoder: Decoder) throws {
		let container = try decoder.container(keyedBy: CodingKeys.self)
		let type = try container.decode(String.self, forKey: ._type)
		if type != "file" {
			throw SanityType.SanityDecodingError.invalidType(type: type)
		}

		let asset = try container.decode(SanityType.Ref.self, forKey: .asset)

		self.init(asset: asset, _type: "file")
	}


}


public extension SanityClient {
	func fileURL(_ file: SanityType.File) -> URL? {
		
		let id = file.asset._ref
		
		let comp = id.components(separatedBy: "-")
		
		if comp.count != 3 {
			return nil
		}
		let str = "https://cdn.sanity.io/files/\(self.config.projectId)/\(self.config.dataset)/\(comp[1]).\(comp[2])"
		return URL(string: str)

	}
}

```

**Comment by sgulseth** [comment]
Thanks for suggesting, it makes total sense! Also great that you found a workaround the implementation looks good. I'll see if I can get it implemented soon.

**Comment by sgulseth** [comment]
Released in [0.13.0](https://github.com/sanity-io/swift-sanity/tree/0.13.0)
